src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -282,6 +282,34 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     executeOrResume(backendCall, runIdForResumption = runId)
   }
 
+  def useCachedCall(avoidedTo: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = Future {
+    val log = workflowLoggerWithCall(backendCall)
+    authenticateAsUser(backendCall.workflowDescriptor) { gcs =>
+      Try(gcs.copyPrefix(avoidedTo.callGcsPath, backendCall.callGcsPath, Option(log))) match {
+        case Success(attempts) if attempts.forall(_.isSuccess) =>
+          postProcess(backendCall) match {
+            case Success(outputs) => SuccessfulExecutionHandle(outputs, backendCall.downloadRcFile.get.stripLineEnd.toInt, backendCall.hash)
+            case Failure(ex: AggregatedException[_]) if ex.exceptions.map(_.exception).isInstanceOf[SocketTimeoutException] =>
+              // TODO: What can we return here to retry this operation?
+              val error = "Socket timeout occurred in evaluating one or more of the output expressions"
+              log.error(error, ex)
+              FailedExecutionHandle(new Throwable(error, ex))
+            case Failure(ex) => FailedExecutionHandle(ex)
+          }
+        case Success(attempts) =>
+          TryUtil.sequence(attempts.toSeq) match {
```

Comment:
Thanks for the super detailed example... though honestly I'm really not a huge fan of the scalaz stuff, I restructured the code to be as concise as your example but not using scalaz.... what do you think?

Also I get why there's hate for `TryUtil.retryBlock` but why the hate for `TryUtil.sequence`?  That seems super useful and we don't have to move away from the standard library as much.  I'm not convinced that the scalaz equivalent to `TryUtil.sequence` is any better (at least in this particular case)